src/main/java/hudson/remoting/ExportTable.java

```diff
@@ -360,7 +360,7 @@ boolean isRecording() {
     synchronized <T> int export(@Nonnull Class<T> clazz, @CheckForNull T t, boolean notifyListener) {
         if(t==null)    return 0;   // bootstrap classloader
 
-        Entry e = reverse.get(t);
+        Entry<T> e = (Entry<T>) reverse.get(t);
```

Comment:
This cast is safe, as can be seen by reading the surrounding code that places entries into the map. The keys of the map are of type `Object` and the values are of type `Entry`. The only time that we ever add something to the map, the `Object` is parameterized on `T` and the `Entry` is also parameterized on the same `T`.